engine/execution/state/state.go

```diff
@@ -286,21 +286,19 @@ func (s *state) SaveExecutionResults(
 	header := result.ExecutableBlock.Block.Header
 	blockID := header.ID()
 
+	err := s.chunkDataPacks.Store(result.AllChunkDataPacks())
+	if err != nil {
+		return fmt.Errorf("can not store multiple chunk data pack: %w", err)
+	}
```

Comment:
Do we need to rollback `chunkDataPacks` commit if the following batch commit fails?

Reply:
Yes, added the rollback

Reply:
Thanks for adding the `chunkDataPacks` rollback.

I took a closer look and it seems that there are three db operations:
- batch write for chunk data packs
- batch write for execution results
- update for HighestExecutedBlock

If any of the db operation fails, do we want to rollback previous operations?  For example, if update fails in the last step, do we want to rollback the previous two batch writes?  What is the expected data consistency?

Reply:
Good point. 

If the second db operation succeeded, the block is considered as executed, because we check whether the state commitment exists to determine if a block has been executed, which is saved in the second db operation. So the data is eventually consistent, if the second db operation fails, we make best effort to rollback the first operation. After restart, we will re-execute the block, and if both db operation succeed, then the data is consistent again. 

The last db operation updates the highest executed block, which is currently only used for metrics and logging. If it fails, it's not necessary to rollback the first two operations. So it means in the worst case, the metrics might be off by 1 block, but it wouldn't cause data consistency problem. 

I refactored it so now the rollback only happens when the second db operation fails. If the last step fails, we don't rollback.